hawkular-datamining-api/src/main/java/org/hawkular/datamining/api/base/InMemorySubscriptionManager.java

```diff
@@ -74,8 +67,9 @@ public void setPredictionListener(PredictionListener predictionListener) {
     }
 
     @Override
-    public void subscribe(Subscription subscription) {
+    public synchronized void subscribe(Subscription subscription) {
```

Comment:
Also, maybe, I'd explain why you want to synchronize on a concurrent hashmap (or in your current case on `this`). I think the reason is that you want to avoid the possibility that if 2 threads subscribe on a tenant that had no subscriptions, 1 of the subscriptions could get lost, but I am not sure that was the only purpose.